server-utilities/src/test/java/org/jboss/eap/qe/common/utilities/WaiterTestCase.java

```diff
@@ -0,0 +1,27 @@
+package org.jboss.eap.qe.common.utilities;
+
+import org.jboss.eap.qe.microprofile.common.utilities.Waiter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public class WaiterTestCase {
+
+    @Test
+    public void testWaitingForConditionSucceeds() throws InterruptedException {
+        final Date timeAtStart = new Date();
+
+        final boolean conditionSatisfied = Waiter.waitFor(() -> new Date().getTime() - timeAtStart.getTime() > 4000, 5, TimeUnit.SECONDS);
+
+        Assert.assertTrue("Condition was not satisfied within time frame!", conditionSatisfied);
+    }
+
```

Comment:
What if i add timeout 200ms? 
I know that Waiter.java does `Thread.sleep(500);` ...

Is there a minimal timeout this util can handle ?

Reply:
It depends on how long the condition check will take and when and if the condition will be satisfied. Generally speaking if the condition is not satisfied within first check, timeout lower than ~500 ms doesn't make sense. The delay can be made configurable. I am even thinking of fluent builder API.